ml/main.py

```
@@ -26,17 +25,24 @@ def read_root():
-    )
+    raw = await audio_file.read()
+
+    print(f"Received {len(raw)} bytes, content_type={audio_file.content_type}")
+    if len(raw) < 100:
+        raise HTTPException(status_code=400, detail="Audio payload too small")
+
+    sample_rate = 16000
+    try:
+        audio_array = decode_audio_bytes(raw, sr=sample_rate)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=f"Could not decode audio: {e}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)